### PR TITLE
Add border controls to Cover and Featured Image blocks

### LINKED ISSUE #37757: Add border controls to the blocks that are similar to Image
In my theme I've added a default border radius to Image blocks. It would be nice to have the option to do this for other blocks that appear in a similar way to images. IE, add border controls to:

- [x] Featured Image
- [ ] Site logo
- [x] Cover
- [ ] Embed
- [x] Comment Author Avatar

### PR DESCRIPTION
**Pull request by jameskoster**
Screenshot 2022-01-07 at 16 20 52

This PR makes it possible to add border radii, color, width, and style to the following blocks:

* Cover
* Featured Image

### To test Cover

* Insert a Cover block
* Add border options like color, width, radius, style
* Check that the correct values are rendered in the Editor and on the frontend
* Try adding image and video backgrounds, then check everything still renders correctly

### To test Featured Image

* Edit a post without a featured image set
* Insert the Featured Image block
* Add border options like color, width, radius, style
* Check that the featured image placeholder renders those options on the canvas
* Upload (or select from media library) a featured image
* Ensure the options still render correctly, both on the canvas and on the frontend

### CODE REVIEW

**Comment by ndiego** [comment]
@jameskoster would it be possible to break this PR into three separate PRs, one for each block? The Featured Image and Cover block implementations seem pretty straightforward, leaving just the Site Logo for further discussion.

**Comment by jameskoster** [comment]
I reverted the Site Logo so we can concentrate on that one separately.

File: packages/block-library/src/post-featured-image/block.json [hunk 1]
```diff
@@ -37,6 +37,18 @@
...
+			"color": true,
+			"radius": true,
+			"style": true,
+			"width": true,
+			"__experimentalDefaultControls": {
+				"color": false,
+				"radius": false,
+				"style": false,
+				"width": false
+			}
```
Comment (aaronrobertshaw) [inline packages/block-library/src/post-featured-image/block.json:51]: If all the border controls are in fact to be optional, I think you can omit this part of the configuration.

That said, when the border support panel was first switched to the `ToolsPanel`, the general idea was that any block opting into more than the border radius support should have all its border support features as default controls. 

From memory, this was primarily to address UX issues around the split border color, width, and style controls. The fact this clutters the sidebar was also a driving force in refining the border controls and delaying opting into border support more widely.

Some background and context here can be found in https://github.com/WordPress/gutenberg/pull/34061 and https://github.com/WordPress/gutenberg/pull/36942.

**Review by aaronrobertshaw — CHANGES_REQUESTED** [review]
Thanks for putting this together @jameskoster 👍 

It might interest you that the Cover block portion of this PR appears to be a duplicate of https://github.com/WordPress/gutenberg/pull/31370. 

The [linked PR](https://github.com/WordPress/gutenberg/pull/31370) was put on the back burner until new color controls were implemented and the border supports refined. It was delayed further when it was decided that the UX around separate controls for border color, width, and style, wasn't great and we should avoid rolling out the support until that was addressed.

There is ongoing work towards refining the border controls into a single component and these PRs should be merged in the near future ( https://github.com/WordPress/gutenberg/pull/37769, https://github.com/WordPress/gutenberg/pull/37770, https://github.com/WordPress/gutenberg/pull/38876 )

My suggestion would be to continue with @ndiego's idea of splitting this PR into one per block, i.e. remove the Cover block changes here.

---

I did also give this PR a test and found the border radius isn't applied cleanly.

This is what I see for the cover block:

Screen Shot 2022-03-16 at 8 54 52 pm

The same issue is present for the Featured Image block also.

Screen Shot 2022-03-16 at 9 04 56 pm

**Comment by jameskoster** [comment]
Closing in favor of https://github.com/WordPress/gutenberg/pull/42765

**Pull request closed without merging**